experiments/pyexps/tcp_congestion_control.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2021 Max Planck Institute for Software Systems, and
+# National University of Singapore
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+# CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+# SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import simbricks.orchestration.experiments as exp
+import simbricks.orchestration.nodeconfig as node
+import simbricks.orchestration.simulators as sim
+from simbricks.orchestration.simulator_utils import create_tcp_cong_hosts
+
+# iperf TCP_multi_client test
+# naming convention following host-nic-net-app
+# host: qemu/gem5-timing
+# nic:  cv/cb/ib
+# net:  switch/dumbbell/bridge
+# app: DCTCPm
+
+#types_of_host = ['qemu', 'qt', 'gt', 'gO3']
+types_of_host = ['gt']
+#types_of_nic = ['cv', 'cb', 'ib']
+types_of_nic = ['ib']
+#types_of_net = ['dumbbell']
+#types_of_app = ['DCTCPm']
+#types_of_mtu = [1500, 4000, 9000]
+types_of_mtu = [1500]
+types_of_congestion_control = ['cubic']
+
+num_pairs = 2
+#max_k = 199680
+#k_step = 16640
+#k_step = 33280
+link_rate_opt = '--LinkRate=200Mb/s'
+link_latency_opt = '--LinkLatency=10ms'
+bdp = 40000000  # Bandwidth-delay product in bytes
+cpu_freq = '5GHz'
+cpu_freq_qemu = '2GHz'
+sys_clock = '1GHz'  # if not set, default 1GHz
+
+ip_start = '192.168.64.1'
+
+experiments = []
+
+# set network sim
+NetClass = sim.NS3DumbbellNet
+
+for congestion_control in types_of_congestion_control:
+    for mtu in types_of_mtu:
+        for host in types_of_host:
+            for nic in types_of_nic:
+                for k_val in range(0, 1 + 1):
```

Comment:
Could use https://docs.python.org/3/library/itertools.html#itertools.product instead to avoid this deep of a nesting.